pkg/pipeline/sink/uploader/uploader.go

```diff
@@ -89,6 +94,17 @@ func (u *noOpUploader) upload(localFilepath, _ string, _ types.OutputType) (stri
 	return localFilepath, stat.Size(), nil
 }
 
+func (u *noOpUploader) cleanupFile(localFilepath string) error {
+	return nil
+}
+
+type baseUploader struct {
+}
+
+func (u *baseUploader) cleanupFile(localFilepath string) error {
```

Comment:
`func (u *Uploader) cleanupFile(localFilepath string) error {`

Reply:
Wouldn't the cause the file to be deleted as well in cases where there is no cloud upload (which I believe means that the noopUploader is selected)?

Reply:
oh, true. In that case, better to kill `*Uploader` and change the `uploader` interface to be public. 
Then the backup logic can move into `baseUploader`